### PR TITLE
Auto-label PRs based on their content [skip-ci]

### PR DESCRIPTION
**Pull request by jolorunyomi**
This PR adds a GitHub action (PR Labeler) to auto label PRs based on a labeling rule and changes in PR.

### CODE REVIEW

**Comment by ajschmidt8** [comment]
Is this a duplicate of #312?

**Review by BradReesWork — APPROVED** [review]

File: .github/labeler.yml [hunk 1]
```diff
@@ -6,20 +6,15 @@ Python:
...
   - '**/CMakeLists.txt'
   - '**/cmake/**'
-  
-Java:
-  - java/**
-  
-Ops:
-  - .github/**
+
+ci:
```
Comment (ajschmidt8) [inline .github/labeler.yml:16]:
```suggestion
gpuCI:
```

File: .github/labeler.yml [hunk 2]
```diff
@@ -6,20 +6,15 @@ Python:
   - python/**
   - notebooks/**
   
-libcusignal:
+cpp:
```
Comment (ajschmidt8) [inline .github/labeler.yml:9]: This repo has no `cpp` label.

**Review by ajschmidt8 — CHANGES_REQUESTED** [review]

File: .github/labeler.yml (hunk 2 above)
Comment (jolorunyomi) [reply .github/labeler.yml:9]: Do you have an idea what I can update it to?

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**